node/network/protocol/src/peer_set.rs

```diff
@@ -57,7 +57,8 @@ impl PeerSet {
 				notifications_protocol: protocol,
 				max_notification_size,
 				set_config: sc_network::config::SetConfig {
-					in_peers: 25,
+					// we want our gossip subset to always include reserved peers
+					in_peers: super::MIN_GOSSIP_PEERS as u32 / 2,
```

Comment:
why `/ 2`?

Reply:
We could also make it 2/3 or something. Does not really matter, but it would be good to be lower than MIN_GOSSIP_PEERS.

Reply:
that was suggested by @eskimor, the idea is that if all of the in_peers are occupied by random/malicious nodes, we will still have some reserved nodes among the selected subset